servant-examples/auth-combinator/auth-combinator.hs

```diff
@@ -56,25 +49,54 @@ newtype PublicData = PublicData { somedata :: Text }
 
 instance ToJSON PublicData
 
+-- | Our private API that we want to be auth-protected.
+type PrivateAPI = Get '[JSON] [PrivateData]
+
+-- | Our public API that doesn't have any protection
+type PublicAPI = Get '[JSON] [PublicData]
+
+-- | Our API, with auth-protection
+type API = "private" :> AuthProtect "cookie-auth" :> PrivateAPI
+      :<|> "public"  :> PublicAPI
+
+-- | A value holding our type-level API
 api :: Proxy API
 api = Proxy
 
+-- | We need to specify the data returned after authentication
+type instance AuthReturnType (AuthProtect "cookie-auth") = User
```

Comment:
For `BasicAuth` you don't need a `usr` parameter or type families, right? Since we know what parameter types are going to be used for the server and the client.